### PR TITLE
[feat]: [CDS-87899] : services v1 yaml schema

### PR DESCRIPTION
**Pull request by tchaurasiya**
pseudo structure
```
DeploymentStageConfig
    services:
        values:
            - ServiceYaml or runtimeInput
        inherit:  (in place of values)
```

```
ServiceYaml
    ref: (required)
    inputs: json or input node
```

### CODE REVIEW

File: v1/pipeline/stages/cd/service-yaml.yaml [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-- identifier
-- name
+  - ref
 properties:
-  description:
+  inputs:
+    oneOf:
+      - $ref: ../../common/json-node.yaml
+      - type: string
+        pattern: ^<\+inputs\..*>$
```
Comment (archit-harness) [inline v1/pipeline/stages/cd/service-yaml.yaml:10]: pattern is not required

File: v1/pipeline/stages/cd/service-yaml.yaml [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
 type: object
 required:
-- identifier
-- name
+  - ref
 properties:
-  description:
+  inputs:
+    oneOf:
+      - $ref: ../../common/json-node.yaml
```
Comment (archit-harness) [inline v1/pipeline/stages/cd/service-yaml.yaml:8]: @shalini-agr are we taking like this jsonNode?

Comment (shalini-agr) [reply v1/pipeline/stages/cd/service-yaml.yaml:8]: we should use this instead
`inputs:
    type: object
    description: inputs to be used in the linked template
    additionalProperties:
      anyOf:
        - string
        - array
        - number
        - object
        - boolean
    propertyNames:
      pattern: "^[a-zA-Z_][0-9a-zA-Z_\\.$-]{0,127}$"`

File: v1/pipeline/stages/cd/inherit-service.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+title: InheritService
+type: object
+required:
+  - stage
+properties:
+  stage:
+    type: string
+    pattern: ^[a-zA-Z_][0-9a-zA-Z_]{0,127}$
+  description:
```
Comment (shalini-agr) [inline v1/pipeline/stages/cd/inherit-service.yaml:9]: property name should be desc

File: v1/pipeline/stages/cd/inherit-service.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    desc: This is the description for InheritService
```
Comment (shalini-agr) [inline v1/pipeline/stages/cd/inherit-service.yaml:10]: this should be description

Comment (archit-harness) [reply v1/pipeline/stages/cd/inherit-service.yaml:10]: it should be this

desc:
   description:

File: v1/pipeline/stages/cd/service-yaml.yaml [hunk 5]
```diff
@@ -1,18 +1,16 @@
...
-    desc: This is the description for ServiceYaml
-  identifier:
-    type: string
-  name:
-    type: string
-  tags:
-    type: object
-    additionalProperties:
-      type: string
+    pattern: ^(?=\s*\S).*$
```
[the remark this answers is not in the document]
Comment (tchaurasiya) [reply v1/pipeline/stages/cd/service-yaml.yaml:13]: @archit-harness this is minimalistic pattern to not to have at least one not whitespace character.
Are we following not having this pattern in our code?

File: v1/pipeline/stages/cd/services-yaml.yaml [hunk 6]
```diff
@@ -1,17 +1,23 @@
...
-    $ref: services-metadata.yaml
   values:
     oneOf:
     - type: array
       items:
-        $ref: service-yaml-v2.yaml
+        $ref: service-yaml.yaml
     - type: string
-      pattern: ^<\+input>((\.)((executionInput\(\))|(allowedValues|default|regex)\(.+?\)))*$
+      pattern: ^<\+inputs\..*>$
```
[the remark this answers is not in the document]
Comment (tchaurasiya) [reply v1/pipeline/stages/cd/services-yaml.yaml:10]: I am a bit confused how are we gonna support runtime values on pipeline (or how runform will be created).
Seems better to not have pattern here.
Will discuss with you above tomorrow and remove it.

File: v1/pipeline/stages/cd/service-yaml.yaml (hunk 2 above)
Comment (tchaurasiya) [reply v1/pipeline/stages/cd/service-yaml.yaml]: Thanks @shalini-agr. I think for every entity we are gonna same inputs schema. I have created this file to be used atleast in cd context. cd-inputs.yaml

Comment (archit-harness) [reply v1/pipeline/stages/cd/service-yaml.yaml]: @tchaurasiya @shalini-agr can we create one common ref file for this inputs?

File: v1/pipeline/stages/cd/service-yaml.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (tchaurasiya) [reply v1/pipeline/stages/cd/service-yaml.yaml]: Removed it.

File: v1/pipeline/stages/cd/services-yaml.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (tchaurasiya) [reply v1/pipeline/stages/cd/services-yaml.yaml:10]: discussed offline. keeping this pattern.

File: v1/pipeline/stages/cd/service-yaml.yaml (hunk 1 above)
Comment (tchaurasiya) [reply v1/pipeline/stages/cd/service-yaml.yaml:10]: discussed offline

File: v1/pipeline/stages/cd/service-yaml.yaml (hunk 2 above)
Comment (tchaurasiya) [reply v1/pipeline/stages/cd/service-yaml.yaml]: discussed offline. moved to common package for every ones usages.

**Review by archit-harness — DISMISSED** [review]

**Review by archit-harness — APPROVED** [review]

File: v1/pipeline/common/inputs.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+title: Inputs
+type: object
+description: inputs to be used for cd entities
```
Comment (shalini-agr) [inline v1/pipeline/common/inputs.yaml:3]: can we remove the term cd entities from here as we would be using it in other stages as well?
can we also add 
```
propertyNames:
      pattern: "^[a-zA-Z_][0-9a-zA-Z_\\.$-]{0,127}$"
```
at the end of this file parallel to additionalProperties

**Review by archit-harness — APPROVED** [review]

**Review by shalini-agr — APPROVED** [review]

**Review by brijeshdhakar — APPROVED** [review]

**Pull request merged**